plugin/storage/factory.go

```diff
@@ -131,3 +131,29 @@ func (f *Factory) InitFromViper(v *viper.Viper) {
 		}
 	}
 }
+
+// CreateArchiveSpanReader implements storage.ArchiveFactory
+func (f *Factory) CreateArchiveSpanReader() (spanstore.Reader, error) {
+	factory, ok := f.factories[f.SpanStorageType]
```

Comment:
this means that the span store and archive span store both have to be using the same storage type? I think that's a fair assumption

Reply:
it can be changed in the future by introducing another env var.